test/common/upstream/health_check_fuzz.proto

```diff
@@ -13,8 +13,13 @@ message HttpRespond { //TODO: Split this across headers, body, and trailers?
   string status = 2 [(validate.rules).string = {pattern: "[0-9]+$*", min_len: 1}];
 }
 
+message TcpRespond {
+  bytes data = 1;
+}
+
 message Respond {
   HttpRespond http_respond = 1 [(validate.rules).message.required = true];
+  TcpRespond tcp_respond = 2 [(validate.rules).message.required = true];
```

Comment:
Hey Adi, me and Asra discussed this design question in length :)! So there comes a part in the action sequence where there has to be a logical difference in the three types. Due to them all implementing from health_checker_impl_base, the events can be shared amogst the three types. However, the only logical difference comes from what the health checkers recieve as a response. Http mainly receives Response Header maps (although it does take in Metadata and 100 continue headers, which I plan to add after I finish gRPC fuzzing), TCP takes in raw bytes, and GRPC takes in Response Header Maps, Raw bytes (which it converts to a gRPC frame), and Trailers. Thus, me and Asra decided to be able to use the action sequence for all three, it would be best to split them in the action sequence, so we don't have to make seperate action streams for each of the three types. Requiring all three responses means that for any generated type, it can simply switch across the type and use that response generated for each of the three specific types of health checkers. However, since this is a very important design question (seems like it'll affect efficiency a lot), let me know if you have a better solution in mind! Thanks! I'll add this as a section in my design doc.

Reply:
> Http mainly receives Response Header maps (although it does take in Metadata and 100 continue headers, which I plan to add after I finish gRPC fuzzing), TCP takes in raw bytes, and GRPC takes in Response Header Maps, Raw bytes (which it converts to a gRPC frame), and Trailers.

I think there's two types of options to weigh:
(1) generate all types of response vs generate one type of response
(2) have three separate response types vs one single response type that's converted into the appropriate

Right now you have generate all types and have 3 separate response types. 
You could also generate required types (1) headers, (2) body (either raw byte data or grpc frame), (3) trailers that at least allows any one of the types to be run (and metadata/100 continue be optional). The code would have to select and convert them to the appropriate type of health checker. That means a little more overlap. It also might be a little strange for grpc. 

The other alternative is have a oneof each type of respond, and simple ignore them if they don't line up. I think this is too much waste-age over generating all 3.

Reply:
I see what you mean.
I agree that generating all 3 should be ok, as long as the fuzzer doesn't spend much time trying different Type-A configs when the configuration is set to Type-B. Let's keep the 3 options, and we can change it later if we think the fuzzing will benefit from it.